module/builder/consensus/builder.go

```diff
@@ -22,19 +22,20 @@ import (
 // Builder is the builder for consensus block payloads. Upon providing a payload
 // hash, it also memorizes which entities were included into the payload.
 type Builder struct {
-	metrics   module.MempoolMetrics
-	tracer    module.Tracer
-	db        *badger.DB
-	state     protocol.MutableState
-	seals     storage.Seals
-	headers   storage.Headers
-	index     storage.Index
-	blocks    storage.Blocks
-	resultsDB storage.ExecutionResults
-	guarPool  mempool.Guarantees
-	sealPool  mempool.IncorporatedResultSeals
-	recPool   mempool.ExecutionTree
-	cfg       Config
+	metrics    module.MempoolMetrics
+	tracer     module.Tracer
+	db         *badger.DB
+	state      protocol.MutableState
+	seals      storage.Seals
+	headers    storage.Headers
+	index      storage.Index
+	blocks     storage.Blocks
+	resultsDB  storage.ExecutionResults
+	receiptsDB storage.ExecutionReceipts // to persist received execution receipts
```

Comment:
❗ **Caution** I added this member field but I haven't changed to constructor to inject a proper value (would require updating also main). Could you do this (?) 🙏 🙇